src/main.rs

```diff
@@ -551,6 +825,48 @@ fn main() {
                 }
             }
 
+            if  device_config.is_audio_device().unwrap_or_default(){
+                
+                if taking_test_recoding && (Instant::now() - last_state_read).as_millis() > 1000{
```

Comment:
Doing any kind of I2C comms blocks for potentially a long time – is it safe to do it here in this loop?  Maybe it doesn't matter as it only happens when a test recording is being taken?